Mathlib/NumberTheory/LSeries/Linearity.lean

```diff
@@ -132,3 +130,57 @@ lemma LSeriesSummable.smul_iff {f : ℕ → ℂ} {c s : ℂ} (hc : c ≠ 0) :
 @[simp]
 lemma LSeries_smul (f : ℕ → ℂ) (c s : ℂ) : LSeries (c • f) s = c * LSeries f s := by
   simp only [LSeries, term_smul_apply, tsum_mul_left]
+
+/-!
+### Sums
+-/
+
+section sum
+
+variable {ι : Type*} [DecidableEq ι] (f : ι → ℕ → ℂ) (S : Finset ι) (s : ℂ)
+
+lemma LSeries.term_sum_apply (n : ℕ) :
+    term (∑ i ∈ S, f i) s n  = ∑ i ∈ S, term (f i) s n := by
+  induction S using Finset.induction_on with
+  | empty =>
+    simp only [Finset.sum_empty, term, Pi.zero_apply, zero_div, ite_self]
+  | insert hi IH  =>
+    simp only [Finset.sum_insert hi, term_add_apply, IH]
+
+lemma LSeries.term_sum : term (∑ i ∈ S, f i) s  = ∑ i ∈ S, term (f i) s :=
+  funext fun _ ↦ by rw [Finset.sum_apply]; exact term_sum_apply f S s _
+
+variable {f S s}
+
+lemma LSeriesHasSum.sum {a : ι → ℂ} (hf : ∀ i ∈ S, LSeriesHasSum (f i) s (a i)) :
+    LSeriesHasSum (∑ i ∈ S, f i) s (∑ i ∈ S, a i) := by
+  simpa only [LSeriesHasSum, term_sum, Finset.sum_fn S fun i ↦ term (f i) s] using hasSum_sum hf
+
+lemma LSeriesSummable.sum (hf : ∀ i ∈ S, LSeriesSummable (f i) s) :
+    LSeriesSummable (∑ i ∈ S, f i) s := by
+  simpa only [LSeriesSummable, ← term_sum_apply] using summable_sum hf
+
+@[simp]
+lemma LSeries_sum (hf : ∀ i ∈ S, LSeriesSummable (f i) s) :
+    LSeries (∑ i ∈ S, f i) s = ∑ i ∈ S, LSeries (f i) s := by
+  simpa only [LSeries, term_sum, Finset.sum_apply] using tsum_sum hf
+
+variable [Fintype ι]
+
+/-- The version of `LSeriesHasSum.sum` for `Fintype.sum`. -/
+lemma LSeriesHasSum.sum' {a : ι → ℂ} (hf : ∀ i, LSeriesHasSum (f i) s (a i)) :
```

Comment:
I think `Fintype.sum` doesn't actually exist, and `∑ i : ι, f i` is just syntactic sugar for `∑ i ∈ univ, f i`. So I am not sure what these additional lemmas are needed for.

Reply:
There are quite a few lemmas called `Fintype.sum_<blah>`, so there is precedent. But you are right that `Fintype.sum` as such does not exist. So the question is perhaps if the existence of the other `Fintype.sum_<blah>` lemmas justifies having these variants here, too. (The docstrings need to be changed, of course.)

Reply:
(At the one point where I'm using the `Fintype` variant later, I can of course easily switch to the unprimed version at the cost of adding a `_`, so I'm open to removing these.)

Reply:
I recall this discussion came up on another PR recently – I can't remember if that was one of yours or someone else's – and we agreed that there is only a need for a `Fintype.sum_xxx` variant when the fintype statement was genuinely different from, and simpler than, the corresponding `Finset` statement applied to `univ`, as with `Fintype.prod_sum` compared to `Finset.prod_sum`. By your last message it sounds like that isn't the case here.